share/spack/gitlab/cloud_pipelines/configs/ci.yaml

```diff
@@ -22,6 +22,10 @@ ci:
       script::
       - spack --color=always --backtrace ci rebuild --tests > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
       after_script:
+      - - spack python ${CI_PROJECT_DIR}/share/spack/gitlab/cloud_pipelines/scripts/common/aggregate_package_logs.spack.py
```

Comment:
This doesn't work for me, I think because any environment variables set in the `script` or `before_script` are not available in `after_script` [(docs)](https://docs.gitlab.com/ee/ci/variables/where_variables_can_be_used.html#execution-shell-environment).

Reply:
All of those vars should be set in the `variables` section, so it should work. Only modifications from the other scripts aren't carried.

Reply:
I think the problematic environment variables are the ones set during the `. share/spack/setup-env.sh` in order to know what the `spack` command is.  But if you tested this and it worked for you, I'd be curious to know what's different about your setup.